python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py

```diff
@@ -53,29 +87,32 @@ async def get_chat_message_contents(
         Returns:
             List[ChatMessageContent]: A list of ChatMessageContent objects representing the response(s) from the LLM.
         """
-        if not settings.ai_model_id:
-            settings.ai_model_id = self.ai_model_id
-        settings.messages = self._prepare_chat_history_for_request(chat_history)
-        settings.stream = False
-        async with (
-            AsyncSession(self.session) as session,
-            session.post(str(self.url), json=settings.prepare_settings_dict()) as response,
-        ):
-            response.raise_for_status()
-            response_object = await response.json()
-            return [
-                ChatMessageContent(
-                    inner_content=response_object,
-                    ai_model_id=self.ai_model_id,
-                    role=AuthorRole.ASSISTANT,
-                    content=response_object.get("message", {"content": None}).get("content", None),
-                )
-            ]
+        settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, OllamaChatPromptExecutionSettings)  # nosec
+
+        prepared_chat_history = self._prepare_chat_history_for_request(chat_history)
+
+        response_object = await AsyncClient(host=self.host).chat(
```

Comment:
Shouldn't we have a persistent client?

Reply:
What is the benefit of having one? The client seems to be a lightweight object.